alerter.c

```diff
@@ -1,19 +1,25 @@
 #include <stdio.h>
 #include <assert.h>
+#include"UnitTest_config.h"
 
 int alertFailureCount = 0;
+int NetworAlertStatus_Test =0;
 
 int networkAlertStub(float celcius) {
     printf("ALERT: Temperature is %.1f celcius.\n", celcius);
     // Return 200 for ok
     // Return 500 for not-ok
     // stub always succeeds and returns 200
-    return 200;
+    return NetworAlertStatus_Test;
 }
 
 void alertInCelcius(float farenheit) {
     float celcius = (farenheit - 32) * 5 / 9;
+    #ifdef UNIT_TEST

```

Comment:
Using conditional compilation for separating production and test code may get complicated as the code size grows.
it is better to use interfaces or functions pointers or lambda functions (if available) to inject either production or test code based on the envt.

Reply:
updated